akka-http-core/src/main/scala/akka/http/impl/settings/ClientConnectionSettingsImpl.scala

```diff
@@ -30,7 +31,8 @@ private[akka] final case class ClientConnectionSettingsImpl(
   websocketRandomFactory:     () ⇒ Random,
   socketOptions:              immutable.Seq[SocketOption],
   parserSettings:             ParserSettings,
-  localAddress:               Option[InetSocketAddress])
+  localAddress:               Option[InetSocketAddress],
+  transport:                  ClientTransport)
```

Comment:
We cannot add another setting at that level because we already have it on `ConnectionPoolSettings` which also wraps `ClientConnectionSettings`. Having it on two layers would be quite confusing. The pattern used for other non-pool client APIs is to add another overload that takes a `ClientTransport` parameter directly.

We might also want to reconsider the decision to put the transport into the pool settings in the first place, instead of putting it into `ClientConnectionSettings` where it could be used by all the APIs.

Reply:
I created https://github.com/akka/akka-http/issues/1562 to track this issue. Maybe you would be interested in doing that change as a preliminary to this one, @JeanFrancoisGuena? Thanks anyway for making us aware of it.

Reply:
I made some change to move `ClientTransport` from `ConnectionPoolSettings` to `ClientConnectionSettings` and pushed.

Unfortunately, I still have a "Binary compatibility check failed!"...